pkg/controller/kubelet-config/helpers.go

```diff
@@ -159,7 +159,7 @@ func getManagedKubeletConfigKey(pool *mcfgv1.MachineConfigPool, client mcfgclien
 		}
 		val, ok := kc.GetAnnotations()[ctrlcommon.MCNameSuffixAnnotationKey]
 		// If we find a matching kubelet config and it is the only one in the list, then return the default MC name with no suffix
-		if !ok && len(kcList.Items) < 2 {
+		if !ok {
```

Comment:
We should add a condition for val so `if !ok || val == ""`. We will need to do the same for the container-runtime-config controller as well.